store/sqlstore/team_store.go

```diff
@@ -381,6 +381,21 @@ func (s SqlTeamStore) GetAllPrivateTeamListing() ([]*model.Team, *model.AppError
 	return data, nil
 }
 
+func (s SqlTeamStore) GetAllPublicTeamPageListing(offset int, limit int) ([]*model.Team, *model.AppError) {
+	query := "SELECT * FROM Teams WHERE AllowOpenInvite = 1 ORDER BY DisplayName LIMIT :Limit OFFSET :Offset"
```

Comment:
Is this supposed to include deleted teams?

Reply:
@grundleborg, good question.  I am following the query from the already existing `GetAllPublicTeamPageListing()` method which doesn't do any accounting for deleted teams.

```go
	query := "SELECT * FROM Teams WHERE AllowOpenInvite = 1 ORDER BY DisplayName LIMIT :Limit OFFSET :Offset"
```

Reply:
@jwilander perhaps you can chime in.  Ultimately, these methods are called from the `getAllTeams` function in the api layer.  Note this function also doesn't include an option for including/excluding deleted teams.

https://github.com/mattermost/mattermost-server/blob/4a60ea23a2f7f2b161e01a40d5b582b3614b0871/api4/team.go#L731

Reply:
I'm OK with consistency for now.